spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCallStatementParser.java

```diff
@@ -68,11 +68,29 @@ public static void stopSpark() {
     currentSpark.stop();
   }
 
+  @Test
+  public void testDelegateUnsupportedProcedure() {
+    assertThatThrownBy(() -> parser.parsePlan("CALL cat.d.t()"))
+        .isInstanceOf(ParseException.class)
+        .hasMessageContaining("[PARSE_SYNTAX_ERROR] Syntax error at or near 'CALL'.");
```

Comment:
In 4.0 we won't need this correct? One of the things I don't like about this (but I don't think there is a fix in 3.5) is that Parse errors are very uninformative.

Reply:
Spark already has error class in 3.5. Shall we do something like
```
    assertThatThrownBy(() -> parser.parsePlan("CALL cat.d.t()"))
        .isInstanceOf(ParseException.class)
        .satisfies(exception -> {
            ParseException parseException = (ParseException) exception;
            assertThat(parseException.getErrorClass()).isEqualTo("PARSE_SYNTAX_ERROR");
            assertThat(parseException.getMessageParameters().get("error")).isEqualTo("'CALL'");
         });
```

Reply:
> In 4.0 we won't need this correct? One of the things I don't like about this (but I don't think there is a fix in 3.5) is that Parse errors are very uninformative.

@RussellSpitzer Yes, with SPARK-44167, Spark supports to parse the `CALL` statement, and a `FAILED_TO_LOAD_ROUTINE` error class will be thrown for this case.

@huaxingao thank you for information, will change soon.